### DISCUSSION TITLE
engine.io-client instantly permanently attempt to reconnect?

### DISCUSSION THREAD

**Discussion by tgbv** [socketio/socket.io#3995] [Q&A] [answered]
I'm using the latest socketio versions (4.0)

I have a WS server which is not always online. Is there a way to instantly permanently attempt to reconnect to it in case transport errors/connection refused are encountered? 

Default settings don't seem to work that way: if the server is down, socket.io-client gives up reconnect.

**Comment by darrachequesne** [comment, marked as answer]
Hi! By default, the Socket.IO client will try to reconnect forever. Available options:

- `reconnection` (defaults to `true`)
- `reconnectionAttempts` (defaults to `Infinity`)
- `reconnectionDelay` (defaults to `1000`)
- `reconnectionDelayMax` (defaults to `5000`)
- `randomizationFactor` (defaults to `0.5`)

Documentation: https://socket.io/docs/v4/client-initialization/#reconnection

> engine.io-client instantly permanently attempt to reconnect?

The Engine.IO client will not try to reconnect. It behaves like a classic WebSocket client, which means you'll have to manually reconnect. The reconnection is a feature of the Socket.IO client above.
 
> I have a WS server

Please note the neither the Engine.IO client nor the Socket.IO client will be able to reach a plain WebSocket server.

Documentation: https://socket.io/docs/v4/#What-Socket-IO-is-not
